src/prototype.rs

```diff
@@ -318,6 +429,26 @@ impl<'a> FirstPass<'a> {
         }
     }
 
+
+    /// Appends a line of HTML to the tree.
+    fn append_html_line(&mut self, start: usize, end: usize)
```

Comment:
Suggestion, this could take a remaining_space argument, as does the `append_code_text`.